dask/array/core.py

```diff
@@ -1141,7 +1149,22 @@ def __getitem__(self, index):
         dsk, chunks = slice_array(out, self.name, self.chunks, index2)
 
         graph = HighLevelGraph.from_collections(out, dsk, dependencies=[self])
-        return Array(graph, out, chunks, dtype=self.dtype)
+        if isinstance(index2, tuple):
+            new_index = []
+            for i in range(len(index2)):
+                if not isinstance(index2[i], tuple):
+                    types = [int, np.int_, list, np.ndarray]
```

Comment:
Can replace `int` and `np.int_` with [`Integral`]( https://docs.python.org/3/library/numbers.html#numbers.Integral ) from the `numbers` module.

Reply:
Thanks for the suggestion. Done.